crates/e2e/tests/e2e/smart_contract_orders.rs

```diff
@@ -157,7 +156,7 @@ async fn smart_contract_orders(web3: Web3) {
         .call()
         .await
         .expect("Couldn't fetch native token balance");
-    assert_eq!(balance, U256::from(7_975_363_406_512_003_608_u128));
+    assert!(balance > to_wei(6));
```

Comment:
This assertion changed pretty significantly. Did switching to fee=0 change something fundamentally in this test?

Reply:
The change is that, instead of using the signed fee_amount = 1, solver decides the fee, which means `7_975_363_406_512_003_608_u128 ` changes to something else. Instead of again defining some random number, I decided to provide a proper assertion that should have been used in the first place. We have 2 tests with buy_amount = 3, which means balance needs to be >= 6 for a test to be valid.